### ISSUE TITLE
Scan and decode with another lib then use zxing-js to parse the result

### ISSUE THREAD

**Issue by psyzz** [zxing-js/library#426]
Hello,

thanks for this lib,

I'm using react-native and i already have a library to scan a qrcode/barcode and get a string result from this scan, but i'm looking for a lib to parse the string result like the zxing lib do here : https://zxing.org/w/decode.jspx

Example with a vCard type :

![zxingResult](https://user-images.githubusercontent.com/1016441/113039121-9c1a7180-9197-11eb-84bd-badee4d2bb4a.png)

What i'm looking for is the "Parsed Result".

Can i use this lib do to that with vCard and other qrcode types ?

Regards

**Comment by odahcam** [comment]
Yes, it's perfectly possible. I'll try to set up an example later.

**Comment by psyzz** [comment]
@odahcam thx! Maybe juste an indication where to look in the code, to avoid you set up an example > Yes, it's perfectly possible. I'll try to set up an example later.

**Comment by psyzz** [comment]
Hello @odahcam , thanks for your time.

Could you point me in the right direction ? I saw ResultParser in the java source code but i didn't find ResultParser in the js library.

Regards,
Jérémy

**Comment by odahcam** [comment]
Hi! Sorry I didn't do anything yet. Could you provide more info on this string's format maybe? Could you post an example string in here as well? I need to find if and what we need to do to pass the data to the library for it to get correctly decoded.

**Comment by psyzz** [comment]
No problem,
An example would be the string i posted on the screenshot, the "Raw text"

**Comment by psyzz** [comment]
Like this one :

BEGIN:VCARD
VERSION:3.0
N:Gump;Forrest
FN:Forrest Gump
ORG:Bubba Gump Shrimp Co.
TITLE:Shrimp Man
PHOTO;VALUE=URL;TYPE=GIF:http://www.example.com/dir_photos/my_photo.gif
TEL;TYPE=WORK;VOICE:(111) 555-1212
TEL;TYPE=HOME;VOICE:(404) 555-1212
ADR;TYPE=WORK:;;100 Waters Edge;Baytown;LA;30314;United States of America
LABEL;TYPE=WORK:100 Waters Edge\nBaytown, LA 30314\nUnited States of America
ADR;TYPE=HOME:;;42 Plantation St.;Baytown;LA;30314;United States of America
LABEL;TYPE=HOME:42 Plantation St.\nBaytown, LA 30314\nUnited States of America
EMAIL;TYPE=PREF,INTERNET:forrestgump@example.com
REV:20080424T195243Z
END:VCARD

**Comment by odahcam** [comment]
So you confirm this is the string result the library you currently use gives you? If so, I'm not sure how ZXing can help, once it would get this information from the image you are scanning/decoding but ZXing is not a VCARD parser.

**Comment by psyzz** [comment]
Yes i confirm this is the string result the library i currently use gives me.
Here is the ResultParser from zxing Java where the result is parsed :
https://github.com/zxing/zxing/blob/master/core/src/main/java/com/google/zxing/client/result/ResultParser.java

As you can see there are many different ResultParser like this one : "SMSMMSResultParser" :

https://github.com/zxing/zxing/blob/d4c3d73592e86aff480afef62ebc20ae70165e38/core/src/main/java/com/google/zxing/client/result/SMSMMSResultParser.java#L41

**Comment by ShimshonMarketing66** [comment]
@psyzz did you find a solution ? if so can you share it please ? 
Thank you
